build.gradle

```diff
@@ -183,6 +118,10 @@ task checksums <<  {
     def checksumsFile = new File("checksums.json")
     checksumsFile.write(JsonOutput.prettyPrint(JsonOutput.toJson(checksums)))
 }
+
+// We order considering that X.Y.Z-SOMETHING should be considered lower than X.Y.Z
+static def orderVersion(str) {
+    return str.split("[.-]").collect { it =~ "^\\d+\$" ? it.padLeft(4,"0") : it.replace("beta","mbeta").replace("BETA","MBETA") }.join(".") + "/"
 }
```

Comment:
It's a bit complicated to understand what this does. Also why does it replace beta with mbeta? Does this consider the release candidates lower that the non-release candidate ones?

Note there were already release candidates published in Maven central: https://mvnrepository.com/artifact/org.neo4j.procedure/apoc

Reply:
Yes it's a little tricky so let me give an explanation and a few examples. Basically this works for both Neo4j and APOC releases. Which usually look something like this:

(**neo4j**)[https://repo1.maven.org/maven2/org/neo4j/neo4j/]
- 1.2.M01
- 3.0.0-M01
- 3.1.0-BETA1      
- 3.2.0-alpha01
- 3.5.0-beta01
- 3.5.0-rc01
- 3.5.1

(**apoc**)[https://repo1.maven.org/maven2/org/neo4j/procedure/apoc/]
- 4.0.0-rc01
- 4.1.0.1

What the function does
- splits these string on "." or "-" character
- if a split is a number then left pad it up to 4 characters
- if a split is a string then it sometimes renames the string others it leaves it as-is

I'm going to remove the string renaming part, because we don't need it for our new project. But if you're curious then it's "beta" to "mbeta" so that the beta versions are listed after the "m01", "m02", etc. versions.

The examples above end up looking like this...

**neo4j** (ordered)
- 0001.0002.M01/
- 0003.0000.0000.M01/
- 0003.0001.0000.BETA1 /
- 0003.0002.0000.alpha01/
- 0003.0005.0000.beta01/
- 0003.0005.0001/
- 0003.0005.0000.rc01/

**apoc** (ordered)
- 0004.0001.0000.0001/
- 0004.0000.0000.rc01/

Ordering is done elsewhere in the code, and it just compares the string's unicode code point, character by character. It's the same way Java does it. As such, "rc" versions appear after non-"rc" versions.